app/lib/LevelSetupManager.coffee

```diff
@@ -92,7 +92,11 @@ module.exports = class LevelSetupManager extends CocoClass
   open: ->
     return @waitingToOpen = true unless @modalsLoaded
     firstModal = if @options.hadEverChosenHero then @inventoryModal else @heroesModal
-    if ((not _.isEqual(lastHeroesEarned, me.get('earned')?.heroes ? []) or
+    if @options.levelID is 'the-gem' && not me.isStudent()
```

Comment:
why not student?

Reply:
interesting, I tried to avoid it because we would need to check if code blocks is in the available format. For teachers and home users, we could hardcode it but for students we would need to add that check. But, you are correct that we should show the modal and probably add a check for it or at least open it without hardcoding it to blocks.